### ISSUE TITLE
bugs when select system.query_log

### ISSUE THREAD

**Issue by youngsofun** [datafuselabs/databend#6758]
```
select written_rows from system.query_log where query_text='insert into  tbl_01_0002 values(1)' and written_rows != 0;
select count(1) from system.query_log where query_text='select count(*) from tbl_01_0002 where a > 1' and scan_partitions = 1 and total_partitions = 2;
```
for now only in ci

1. the first return slowly in mac, and lead to 500 in k3d,  https://github.com/datafuselabs/databend/issues/6614
3. the second return the wrong count sometimes. （including MySQL handler）

cc  @zhang2014 @sundy-li @ZeaLoVe

**Comment by ZeaLoVe** [comment]
https://github.com/datafuselabs/databend/pull/6757   the failed check in this pr relate to these sqls, get no result.

**Comment by sundy-li** [comment]
fixed in #6614
